src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -471,19 +471,25 @@ internal SynthesizedAttributeData SynthesizeDynamicAttribute(TypeSymbol type, in
             Debug.Assert((object)type != null);
             Debug.Assert(type.ContainsDynamic());
 
+            MethodSymbol ctorSymbol;
+            ImmutableArray<TypedConstant> arguments;
+
             if (type.IsDynamic() && refKindOpt == RefKind.None && customModifiersCount == 0)
             {
-                return TrySynthesizeAttribute(WellKnownMember.System_Runtime_CompilerServices_DynamicAttribute__ctor);
+                ctorSymbol = (MethodSymbol)GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_DynamicAttribute__ctor);
+                arguments = ImmutableArray<TypedConstant>.Empty;
             }
             else
             {
-                NamedTypeSymbol booleanType = GetSpecialType(SpecialType.System_Boolean);
-                Debug.Assert((object)booleanType != null);
+                ctorSymbol = (MethodSymbol)GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_DynamicAttribute__ctorTransformFlags);
+                var argumentType = (ArrayTypeSymbol)ctorSymbol.Parameters[0].Type;
+                var booleanType = argumentType.ElementType;
+                Debug.Assert(booleanType.SpecialType == SpecialType.System_Boolean);
                 var transformFlags = DynamicTransformsEncoder.Encode(type, booleanType, customModifiersCount, refKindOpt);
-                var boolArray = ArrayTypeSymbol.CreateSZArray(booleanType.ContainingAssembly, booleanType, customModifiers: ImmutableArray<CustomModifier>.Empty);
-                var arguments = ImmutableArray.Create<TypedConstant>(new TypedConstant(boolArray, transformFlags));
-                return TrySynthesizeAttribute(WellKnownMember.System_Runtime_CompilerServices_DynamicAttribute__ctorTransformFlags, arguments);
+                arguments = ImmutableArray.Create<TypedConstant>(new TypedConstant(argumentType, transformFlags));
             }
+
+            return new SynthesizedAttributeData(ctorSymbol, arguments, ImmutableArray<KeyValuePair<string, TypedConstant>>.Empty);
```

Comment:
What is the motivation for this change? Was this function doing something wrong? Is it safe to assume that the constructor of the right shape will be found? TrySynthesizeAttribute is handling the case of a missing constructor.